### PR TITLE
Correct docker repository for the ingestion server, update docs

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This PR corrects the docker repository name used in production for the ingestion server. This format changed as a part of #437. It also adds a way to easily override the desired tag using an environment variable (in this case, in the local `.env` file on production). This way we can specify which version we want to pull/run without having to update the file on every version bump.

Lastly, I added some documentation to the DNS segment based on discussion in #346.

## Testing Instructions

1. `cd ingestion_server`
1. `docker-compose pull` (optionally setting the `IMAGE_TAG` variable to a specific version tag, e.g. `v2.1.1`)
1. `docker-compose up -d`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: ingestion_server/docker-compose.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
     ports:
       - "8001:8001"
     env_file:
       - .env
     dns:
       # Related to DNS management on the production box.
-      # Not sure if this is strictly necessary, but it exists
-      # in the current, running system so we are retaining it here.
+      # 172.16.0.0–172.31.255.255 is a reserved IP range by the (IANA) for
+      # private/internal networking. This specific IP is the dns resolution address
```
Comment (dhruvkb) [inline ingestion_server/docker-compose.yml:13]:
```suggestion
      # private/internal networking. This specific IP is the DNS resolution address
```

**Review by dhruvkb — APPROVED** [review]
TIL about environment variable interpolation inside `docker-compose.yml`! Very cool 👍

**Comment by AetherUnbound** [comment]
> TIL about environment variable interpolation inside `docker-compose.yml`! Very cool +1

We were trying to come up with a solution on a different project for pinning the version in the prod compose file without having to change the text and this fit the bill perfectly!

**Comment by obulat** [comment]
When running `docker-compose pull`, I am getting this error:
`open ...openverse-api/ingestion_server/.env: no such file or directory` . Do I need to create a `.env` file before running?

**Review by obulat — APPROVED** [review]

**Comment by AetherUnbound** [comment]
@obulat you can also create the `.env` files with `just env`, although you shouldn't need to have them in order to run `docker-compose pull` :thinking: I'll look into that!

**Comment by obulat** [comment]
I also might have had a very old version of api locally...

**Comment by AetherUnbound** [comment]
Oh just kidding, it [absolutely is expecting a `.env` file](https://github.com/WordPress/openverse-api/blob/0e9d2df4c6e3d213c8750ff86b2be18cd19506b9/ingestion_server/docker-compose.yml#L9). This is for use in production so those values will need to be overridden anyway

**Pull request merged**